gtsam/geometry/Cal3Bundler.cpp

```diff
@@ -118,6 +120,25 @@ Point2 Cal3Bundler::calibrate(const Point2& pi, const double tol) const {
     throw std::runtime_error(
         "Cal3Bundler::calibrate fails to converge. need a better initialization");
 
+  const double x = invKPi.x(), y = invKPi.y();
+  const double xx = x * x, yy = y * y;
+  const double rr = xx + yy;
+  const double g = (1 + k1_ * rr + k2_ * rr * rr);
+  const double inv_f = 1 / f_, inv_g = 1 / g;
+
+  // Approximate the jacobians via a single iteration of g.
```

Comment:
Does this comment still hold? You are doing multiple iterations above

Reply:
We're doing multiple iterations but when calculating the jacobians, I assume only one iteration is done. So it neither the initial estimate or the final iterate (which could be after n<=10 iterations), but after a single iteration.

Reply:
What would you recommend adopting? This was one of the reasons this PR took as long as it did.

Reply:
`calibrate` takes undistorted coordinates and makes them into pixels. This is iterative, but at end we have the correct relationship. Then we ask how the answer changes *differentially*, after we found this answer. So, it clearly needs to be the final iterate, as I understand it. So, I'd recommend taking rr, g etc out of the loop, initialize them from x and y, and then update at end of the loop so they are valid for the last estimate.

You still have to document how you got the derivatives. These things are more clear when there is a closed form solution, but it's not obvious when we are iterating on a nonlinear relationship.